### PR TITLE
Check the state of prev_events a bit more thoroughly when coming over federation

### PR DESCRIPTION
**Pull request by hawkowl**

### CODE REVIEW

**Comment by erikjohnston** [comment]
Hmm. I've just had a thought: a server won't necessarily have permission to fetch state for events (e.g. if the server wasn't in the room at the time and history is private).

I guess in the case where an event points to those prev events then we'll discard the event, but pick it up next time someone sends a message?

File: tests/test_federation.py [hunk 1]
```diff
@@ -0,0 +1,243 @@
...
+            ),
+        )
+
+        # Make sure the invalid event isn't there
+        extrem = maybeDeferred(
+            self.homeserver.datastore.get_latest_event_ids_in_room, self.room_id
+        )
+        self.assertEqual(self.successResultOf(extrem)[0], "$join:test.serv")
+
+    @unittest.DEBUG
```
Comment (erikjohnston) [inline tests/test_federation.py]: Was this meant to be left here?

**Comment by erikjohnston** [comment]
I wonder whether we should drop the requirement that you need to be able to produce the prev_events? Given the server may not actually have permission to view the event?

File: tests/test_federation.py (hunk 1 above)
Comment (hawkowl) [reply tests/test_federation.py]: errrr no... but it probably doesn't harm anything for right now :)

**Pull request merged**